### DISCUSSION TITLE
Will watchstate remember the watch state of media that has been removed from plex/jellyfin libraries?

### DISCUSSION THREAD

**Discussion by arcastro** [arabcoders/watchstate#546] [General] [answered]
Hello,

I have both plex and jellyfin pointing to the same media library folders and am interested in using watchstate to sync the state between them.

I am also in the process of cleaning up my media libraries to save on storage, but am concerned about permanently losing the watch state for these items.

If I delete watched movies from my library and add them again at a later point in time, will watchstate remember these items for me and mark them as watched in both plex/jellyfin once they're readded? Or will it go back to being marked as unwatched?

I appreciate the help,

Best regards,

**Comment by arabcoders** [comment, marked as answer]
Hello,

While WatchState will not delete old history by it self, there is many ways to get your self caught into it, for example
* Running data Integrity, and removing the reported records.
* Running file integrity, and removing the reported records.
* Manually removing the record via history tab.

So, with this in mind, for better long term record keeping i'd suggest using external service, However, if you enable backups you should retain the state, however that's not permanent unless you manually trigger the backup as the automatic backup is subject to automatic removal after 7 days.

**Reply by arcastro** [reply]
Thank you @arabcoders, that sounds good. Looks like I'm safe from ever running file integrity, as I don't plan on ever giving watchstate access to the media library folders. I'm also safe from manually removing the records via history tab, since that's something I plan on avoiding.

However, I'm not familiar with the first bullet point, "data integrity". What is that? Where can I find it, or more information about it?

**Reply by arabcoders** [reply]
Hi, sorry `data Integrity` shown as `Data Parity` externally i forgot about that, it's a means to check your backends metadata against each other to see if there is mismatches. i.e. one reporting E1 as E2 etc.

On the file integrity, you can use it safely by mounting your media as read only i.e. `/storage/media:/storage/media:ro` we dont remove anything. and the feature is helpful to see if you have stale records in your media servers. or you want to quickly stream the media to check. However it's your choice =)

**Reply by arcastro** [reply]
Ah, thank you again for the info! that's great! 
I starred the project btw - I deployed it today and so far it's exactly what I was looking for 🙌.

Regarding the file integrity, the main point for opening this question is that I _want_ to keep stale records on watchstate 😄. Meaning if I personally delete a movie from my library, I want to keep the stale record so that it can still be synced to plex in the future if I ever add the movie back again to my media collection. In this case I would still want to keep file integrity disabled, right? Or did I misunderstand the meaning of a stale record?

**Reply by arabcoders** [reply]
As long as you don't click the big red delete button you be ok.
